### PR TITLE
Clang format gpu/segmentation

### PR DESCRIPTION
**Pull request by FabianSchuetze**
After  #4677 got merged, I wanted to do some summer cleaning (or winter cleaning if you read this from the southern hemisphere) in the segmentation codebase. I applied clang-format and did some manual modifications. Everything looks nice and shiny :-)

### CODE REVIEW

File: gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_clusters.h [hunk 1]
```diff
@@ -39,127 +39,178 @@
...
+void
+extractEuclideanClusters(const typename pcl::PointCloud<PointT>::Ptr& host_cloud_,
+                         const pcl::gpu::Octree::Ptr& tree,
+                         float tolerance,
+                         std::vector<PointIndices>& clusters,
+                         unsigned int min_pts_per_cluster,
+                         unsigned int max_pts_per_cluster);
+
+/** \brief @b EuclideanClusterExtraction represents a segmentation class for cluster
+ * extraction in an Euclidean sense, depending on pcl::gpu::octree \author Koen Buys,
```
Comment (kunaltyagi) [inline gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_clusters.h:61]: Fix doxygen please

File: gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h [hunk 2]
```diff
@@ -39,121 +39,173 @@
...
+extractLabeledEuclideanClusters(
+    const typename pcl::PointCloud<PointT>::Ptr& host_cloud_,
+    const pcl::gpu::Octree::Ptr& tree,
+    float tolerance,
+    std::vector<PointIndices>& clusters,
+    unsigned int min_pts_per_cluster,
+    unsigned int max_pts_per_cluster);
+
+/** \brief @b EuclideanLabeledClusterExtraction represents a segmentation class for
+ * cluster extraction in an Euclidean sense, depending on pcl::gpu::octree \author Koen
```
Comment (kunaltyagi) [inline gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h:61]: Doxygen seems to be problematic in a lot of places

**Review by kunaltyagi — CHANGES_REQUESTED** [review]

File: gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h (hunk 2 above)
Comment (FabianSchuetze) [reply gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h:61]: Thanks a lot, Kunal, for the review! Oh boy, clang-format and Doxygen seem to get along with each other as well as cats and dogs do. I tried to make sure that clang-format places the \param keyword on a new line but did not realize that other keywords get misrepresented. I will change that!

File: gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h [hunk 3]
```diff
@@ -100,8 +101,7 @@ class EuclideanLabeledClusterExtraction {
   }
 
   /** \brief Set the spatial cluster tolerance as a measure in the L2 Euclidean space
-   * \param tolerance the spatial cluster tolerance as a measure in the L2 Euclidean
-   * space
+   * \param tolerance the spatial cluster tolerance mesured by L2 distance
```
Comment (kunaltyagi) [inline gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h:104]:
```suggestion
   * \param tolerance the spatial cluster tolerance measured by L2 distance
```

File: gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp [hunk 4]
```diff
@@ -148,69 +148,84 @@ pcl::gpu::extractEuclideanClusters (const typename pcl::PointCloud<PointT>::Ptr
...
-  if(tree_->cloud_.size() != host_cloud.size ())
-  {
-    PCL_ERROR("[pcl::gpu::EuclideanClusterExtraction] size of host cloud and device cloud don't match!\n");
-    return;
-  }
-*/
+  /*
+    if(tree_->cloud_.size() != host_cloud.size ())
+    {
+      PCL_ERROR("[pcl::gpu::EuclideanClusterExtraction] size of host cloud and device
```
Comment (kunaltyagi) [inline gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp:206]: Is this still correct?

File: gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_labeled_clusters.hpp [hunk 5]
```diff
@@ -109,69 +111,82 @@ pcl::gpu::extractLabeledEuclideanClusters (const typename pcl::PointCloud<PointT
...
-  {
-    PCL_ERROR("[pcl::gpu::EuclideanClusterExtraction] size of host cloud and device cloud don't match!\n");
-    return;
-  }
-*/
+  /*
+    if(tree_->cloud_.size() != host_cloud.size ())
+    {
+      PCL_ERROR("[pcl::gpu::EuclideanClusterExtraction] size of host cloud and device
+    cloud don't match!\n"); return;
```
Comment (kunaltyagi) [inline gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_labeled_clusters.hpp:168]: same here

File: gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp (hunk 4 above)
Comment (FabianSchuetze) [reply gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp:206]: Him, this is a good question! This check seems to be active in the corresponding [cpu method](https://github.com/PointCloudLibrary/pcl/blob/d242fcbdbb53efc7de48c9159343432a2194a27c/segmentation/include/pcl/segmentation/extract_clusters.h#L112), but I am not sure this check is needed. I think constructing the tree with a superset of the pointcloud used for clustering should work. Nevertheless, I would prefer to activate it again in the GPU module for the sake of symmetry. What do you think?

File: gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_labeled_clusters.hpp (hunk 5 above)
Comment (FabianSchuetze) [reply gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_labeled_clusters.hpp:168]: see comment  above above

File: gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h (hunk 3 above)
Comment (FabianSchuetze) [reply gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h:104]: You are Eagle-eyed - Thank you!

**Comment by larshg** [comment]
Add it to the script https://github.com/PointCloudLibrary/pcl/blob/d242fcbdbb53efc7de48c9159343432a2194a27c/.dev/format.sh#L11 ?

File: gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp (hunk 4 above)
Comment (kunaltyagi) [reply gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp:206]: @larshg Thoughts?

(separate PR)

**Comment by FabianSchuetze** [comment]
Sorry for the long radio silence - I've been moving place. I added the the folder to the format.sh script. Thanks for pointing that out, @larshg !

**Review by larshg — APPROVED** [review]

**Review by kunaltyagi — APPROVED** [review]

**Comment by kunaltyagi** [comment]
@mvieth Merge? 🚀

**Comment by mvieth** [comment]
No objections from my side

**Pull request merged**